tmt/utils.py

```diff
@@ -6183,3 +6183,29 @@ def __exit__(self, *args: Any) -> None:
     @property
     def duration(self) -> datetime.timedelta:
         return self.end_time - self.start_time
+
+
+def retry(
+        function_to_retry: Callable[..., Any],
+        retry_count: int,
+        delay: int,
+        error_message: str,
```

Comment:
How about not accepting the full error message as a parameter, but the "label" of the action we retried? In the current situation, it'd be "Pulling image", and the rest would be up to `retry:

```
    for _ in range(retries):
        try:
            function_to_retry(*args, **kwargs)
            break
        except Exception:
            print(f'{label} failed, trying again in {delay} seconds.')
            time.sleep(delay)
```

The only part of the error message that changes is the description of what the callable is doing, the rest remains the same, pretty much no matter what the callable would do - we would always want to log what, when...